pkg/plugin/datasource.go

```diff
@@ -28,49 +34,16 @@ func NewMQTTInstance(s backend.DataSourceInstanceSettings) (instancemgmt.Instanc
 	return NewMQTTDatasource(client, s.UID), nil
 }
 
-func getDatasourceSettings(s backend.DataSourceInstanceSettings) (*mqtt.Options, error) {
-	settings := &mqtt.Options{}
-
-	if err := json.Unmarshal(s.JSONData, settings); err != nil {
-		return nil, err
-	}
-
-	if password, exists := s.DecryptedSecureJSONData["password"]; exists {
-		settings.Password = password
-	}
-
-	return settings, nil
-}
-
-type MQTTClient interface {
-	Stream() chan mqtt.StreamMessage
-	IsConnected() bool
-	IsSubscribed(topic string) bool
-	Messages(topic string) ([]mqtt.Message, bool)
-	Subscribe(topic string)
-	Unsubscribe(topic string)
-}
-
 type MQTTDatasource struct {
-	Client        MQTTClient
+	Client        mqtt.Client
 	channelPrefix string
 }
 
-// Make sure MQTTDatasource implements required interfaces.
-// This is important to do since otherwise we will only get a
-// not implemented error response from plugin in runtime.
-var (
-	_ backend.QueryDataHandler      = (*MQTTDatasource)(nil)
-	_ backend.CheckHealthHandler    = (*MQTTDatasource)(nil)
-	_ backend.StreamHandler         = (*MQTTDatasource)(nil)
-	_ instancemgmt.InstanceDisposer = (*MQTTDatasource)(nil)
-)
-
 // NewMQTTDatasource creates a new datasource instance.
-func NewMQTTDatasource(client MQTTClient, uid string) *MQTTDatasource {
+func NewMQTTDatasource(client mqtt.Client, uid string) *MQTTDatasource {
 	return &MQTTDatasource{
 		Client:        client,
-		channelPrefix: fmt.Sprintf("ds/%s/", uid),
+		channelPrefix: path.Join("ds", uid),
```

Comment:
does this always use `/`? on windows does it use `\`

Reply:
`path/filepath` is OS aware, but `path.Join` always uses `/`